### PR TITLE
/whois: Only show semilock reasons to global staff

### PR DESCRIPTION
**Pull request by mia-pi-git**

### CODE REVIEW

**Comment by Zarel** [comment]
Why?

**Comment by Zarel** [comment]
If you're roomstaff and a user can't talk in your room, you should know that.

**Comment by mia-pi-git** [comment]
I changed it to hiding the reasons - is that better?

**Comment by Zarel** [comment]
Sure, but did someone request this? What sorts of reasons did we want to hide from romostaff?

**Comment by mia-pi-git** [comment]
> Sure, but did someone request this? What sorts of reasons did we want to hide from roomstaff?

See #private-dev.

File: server/chat-commands/info.ts [hunk 1]
```diff
@@ -211,7 +211,7 @@ export const commands: ChatCommands = {
 			}
 
 			if (targetUser.semilocked) {
-				buf += `<br />Semilocked: ${targetUser.semilocked}`;
+				buf += `<br />Semilocked: ${user.can('lock') ? targetUser.semilocked : "(reason hidden)"}`;
```
Comment (Zarel) [inline server/chat-commands/info.ts:214]:
```suggestion
				buf += `<br />Semilocked: ${user.trusted ? targetUser.semilocked : "(reason hidden)"}`;
```

I think I prefer a lower threshold like `trusted`. Is that fine with you?

**Comment by Zarel** [comment]
After further staff discussion, it seems most think the `trusted` threshold is too low, so it looks like we're going with this.

Thanks!

**Pull request merged**